cactusbot/commands/magic/config.py

```diff
@@ -16,131 +18,289 @@
     })
 
 
-async def _update_spam_config(api, scope, field, value):
+async def _update_config(api, scope, field, value):
+    """Update a config value."""
+
     return await api.update_config({
         scope: {
             field: value
         }
     })
 
 
+async def _get_event_data(api, event):
+    """Get data about an event."""
+
+    data = (await (await api.get_config()).json())["data"]
+    event = data["attributes"]["announce"][event]
+    return event
+
+
+async def _get_spam_data(api, section):
+    """Get data about a section of spam config."""
+
+    data = (await (await api.get_config()).json())["data"]
+    spam_section = data["attributes"]["spam"][section]
+    return spam_section
+
+
 class Config(Command):
-    """Config command"""
+    """Config Command."""
 
     COMMAND = "config"
 
     @Command.command(role="moderator")
-    class Announce(Command):
-        """Announce sub command."""
+    class Follow(Command):
+        """Follow subcommand."""
 
-        @Command.command()
-        async def follow(self, value):
-            """Follow subcommand."""
+        @Command.command(role="moderator", name="follow")
+        async def default(self, value=""):
+            """Get status, and message of the follow event, or toggle."""
+
+            if not value:
```

Comment:
This is annoying, but currently necessary. Would like to add some form of pattern matching to `command.py` at some point, but for now, this works.